lsp-diagnostics.el

```diff
@@ -202,10 +212,14 @@ provided by lsp-mode.
 See https://github.com/emacs-lsp/lsp-mode."
       :start #'lsp-diagnostics--flycheck-start
       :modes '(lsp-placeholder-mode) ;; placeholder
-      :predicate (lambda () lsp-mode)
+      :predicate (lambda () lsp-diagnostics-mode)
       :error-explainer (lambda (e)
                          (lsp-diagnostics-flycheck-error-explainer
                           e (lsp--workspace-server-id (car-safe (lsp-workspaces)))))))
+  (and lsp-diagnostics--flycheck-enabled
```

Comment:
shouldn't that be `(not  lsp-diagnostics--flycheck-enabled)`

Reply:
yes, fixed now. It worked for me since I only tested with an original value of nil most of the time. Verified that it works enabling/disabling with flycheck-checker actually being set not as well.